examples/mqtt/azure_iot/simulated_home_topic_per_metric_x509_cert.py

```diff
@@ -0,0 +1,228 @@
+#            ----------------                  Light Metric
+#           |                |
+#           |                |
+#      Temperature        Humidity
+#         Metric           Metric
+#
+# Developers can leverage the following options:
+# ---------------------------------------------
+#  a) Use single publish and subscribe topic generated by LIOTA for an EdgeSystem, its Devices and its Metrics.
+#  b) Use custom single publish and subscribe topic for an EdgeSystem, its Devices and Metrics.
+#
+#  - In the above two cases, MQTT message's payload MUST be self-descriptive so that subscriber can subscribe
+#    process accordingly to a single topic by parsing payload.
+#
+#  c) Use custom publish and subscribe topics for Metrics.
+#  - In this case, MQTT message's payload need not be self-descriptive.  Subscribers can subscribe to
+#    appropriate topics and process accordingly.
+#
+#  d) Use combination of (a) and (c) or (b) and (c).
+#
+# GenericMqtt DCC has enclose_metadata option.  It can be used to enclose EdgeSystem, Device and Metric names
+# along with the sensor data payload of a Metric.
+#
+# This example showcases publishing Metrics using (b) and (c) without enclose_metadata
+# ------------------------------------------------------------------------------------------------------------------
+
+if __name__ == '__main__':
+
+    #  Creating EdgeSystem
+    edge_system = Dell5KEdgeSystem(config['EdgeSystemName'])
+
+    #  Encapsulates Credentials
+    credentials = Credentials(config['broker_root_ca_cert'], config["broker_username"],
+                              config["broker_password"], config['edge_system_cert_file'],
+                              config['edge_system_key_file'])
+
+    # Encapsulate TLS parameters
+    tls_conf = TLSConf(config['cert_required'], config['tls_version'], config['cipher'])
+
+    # Encapsulate QoS related parameters
+    qos_details = QoSDetails(config['in_flight'], config['queue_size'], config['retry'])
+
+    #  Connecting to AzureIoT
+    #  Initializing GenericMqtt using MqttDccComms
+    #  Custom Publish Topic for an EdgeSystem
+    #  Azure MQTT support documentation link: https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-mqtt-support
+    mqtt_msg_attr = MqttMessagingAttributes(pub_topic=config['EdgeSystemHealthPubTopic'])
+
+    azure = GenericMqtt(MqttDccComms(edge_system.name, config['BrokerURL'], config['BrokerPort'],
+                                     credentials=credentials, tls_conf=tls_conf, qos_details=qos_details,
+                                     client_id=config["device_id"], clean_session=False, userdata=config['userdata'],
+                                     protocol=config['protocol'], transport=['transport'],
+                                     conn_disconn_timeout=config['ConnectDisconnectTimeout'],
+                                     mqtt_msg_attr=mqtt_msg_attr, enable_authentication=True), enclose_metadata=True)
```

Comment:
Why do we need `enclose_metadata=True` ?  Since it is topic per metric, it should be false.
Also, mqtt_msg_attr should be none